Remove KomenciKit from local repo

gnardini opened the pull request:
### Description

Removed the local komencikit version and imported it from npm. Also added two new parameters which are fetched from Firebase that will allow us to use a new proxy implementation in the future.

### Other changes

N/A

### Tested

Manually verified a phone number in both alfajores and mainnet

### How others should test

Make sure that you can verify your phone number

### Related issues

- Fixes #223

### Backwards compatibility

N/A

tarikbellamine reviewed (commented):
Do we really want to be removing this? I think Valora will eventually be on the hook for maintaining this package.

jeanregisser commented on packages/mobile/test/schemas.ts:
```diff
@@ -702,6 +702,13 @@ export const v16Schema = {
     exchangeRates: null,
     exchangeRatePair: undefined,
   },
+  verify: {
+    ...v15Schema.verify,
+    komenciConfig: {
+      useLightProxy: false,
+      allowedDeployers: [],
+    },
+  },
```
We should add this to a new schema.
Since the last public release already used schema v16, it's safer to make a new one as we might forget about this change later on.

jeanregisser replied:
By the way I missed this in https://github.com/valora-inc/wallet/commit/49d0e31b7621683d716d0ae7613acae3e7c9f273#diff-a0850796be5a07a6363631e739d56ea69fa51c26852655610c6f7650285b9a2e and the new schema should have been added there.

gnardini replied:
Isn't the new schema's use for new migrations? If I add this with no migration both will end up desynced, no?

gnardini replied:
A v17 was introduced in another PR with a migration so I moved this there instead. I don't think I understand the logic for this versioning though :/

jeanregisser reviewed (commented):
Great! 👍

gnardini commented:
> Do we really want to be removing this? I think Valora will eventually be on the hook for maintaining this package.

This package lives in the komenci repo: https://github.com/celo-org/komenci/tree/master/packages/libs/komencikit so it's duplicated atm

gnardini commented:
Since we moved to remote config but remote config doesn't support arrays and komencikit's config requires an array I changed the way we read the config so we still read it from the RTDB rather than remote config.

jeanregisser commented:
> Since we moved to remote config but remote config doesn't support arrays and komencikit's config requires an array I changed the way we read the config so we still read it from the RTDB rather than remote config.

I would prefer we keep them in the Remote Config, we don't want any feature flags remaining in the RTDB moving forward.
What do you think of using comma separated values instead?

jeanregisser reviewed (approved):
LGTM 👍
